plugins/catalog-backend/src/service/types.ts

```diff
@@ -14,12 +14,23 @@
  * limitations under the License.
  */
 
-import { Entity, LocationSpec } from '@backstage/catalog-model';
+import { Entity } from '@backstage/catalog-model';
 import { Location } from '@backstage/catalog-client';
 
+/**
+ * Holds the information required to create a new location in the catalog location store.
+ * @public
+ */
+export interface LocationInput {
+  type: string;
+  target: string;
+  /** @deprecated This field is ignored and will be removed */
+  presence?: 'optional' | 'required';
```

Comment:
We have strict validation of the input on the API endpoint, so figured we can deprecate this rather than remove directly. Another option would be to specifically allow the `presence` field and ignore it, and then remove this immediately, might be cleaner